app/lib/activitypub/activity/add.rb

```diff
@@ -4,9 +4,10 @@ class ActivityPub::Activity::Add < ActivityPub::Activity
   def perform
     return unless @json['target'].present? && value_or_id(@json['target']) == @account.featured_collection_url
 
-    status = status_from_uri(object_uri)
+    status   = status_from_uri(object_uri)
+    status ||= fetch_remote_original_status
```

Comment:
It's… not defined here? How come the tests didn't catch that? Am I missing anything?

Reply:
The tests assume the referenced status was already known. And that's kind of the more likely scenario because only receive Add/Remove if you follow, and if you follow, you get new posts too. I think this case is just of someone pinning an old, pre-follow post.

Reply:
I meant, I `fetch_remote_original_status` is not defined in this part of the code, afaict, I'm surprised the continuous integration tooling haven't caught that, but I understand the tests themselves did not cover this case at all.

Reply:
OH. Oops. I see what you mean. Somehow thought that was defined on the Activity superclass, because I remember that's how it's structured in the OStatus equivalent :weary:

Reply:
@ThibG It's not caught by the test because that line would only be executed if status_from_uri returned nil, which in the test, it does not